nitro-protocol/contracts/ForceMove.sol

```diff
@@ -457,126 +473,79 @@ contract ForceMove is IForceMove, StatusManager {
         Signature[] memory sigs,
         uint8[] memory whoSignedWhat
     ) internal pure returns (bytes32) {
-        bytes32[] memory stateHashes = _requireValidTransitionChain(
-            variableParts,
-            channelId,
-            fixedPart
-        );
+        VariablePart memory latestVariablePart = IForceMoveApp(fixedPart.appDefinition)
+            .latestSupportedState(fixedPart, _bindSigsToVariableParts(variableParts, sigs, whoSignedWhat));
 
-        require(
-            _validSignatures(
-                _lastVariablePart(variableParts).turnNum,
-                fixedPart.participants,
-                stateHashes,
-                sigs,
-                whoSignedWhat
-            ),
-            'Invalid signatures'
-        );
+        // enforcing the latest supported state being in the last slot of the array
+        _requireVariablePartIsLast(latestVariablePart, variableParts);
 
-        return stateHashes[stateHashes.length - 1];
+        return _hashState(
+            channelId,
+            latestVariablePart.appData,
+            latestVariablePart.outcome,
+            latestVariablePart.turnNum,
+            latestVariablePart.isFinal
+        );
     }
 
     /**
-     * @notice Check that the submitted states form a chain of valid transitions
-     * @dev Check that the submitted states form a chain of valid transitions
-     * @param variableParts Variable parts of the states in the support proof
-     * @param channelId Unique identifier for a channel.
-     * @param fixedPart Fixed Part of the states in the support proof
-     * @return true if every state is a validTransition from its predecessor, false otherwise.
+     * @notice Construct SignedVariablePart array from supplied variable parts and signatures.
+     * @dev Construct SignedVariablePart array from supplied variable parts and signatures.
+     * @param variableParts Variable parts to construct signed ones from.
+     * @param sigs Signatures to combine with variable parts.
+     * @param whoSignedWhat participant[i] signed stateHashes[whoSignedWhat[i]]
+     * @return SignedVariablePart array.
      */
-    function _requireValidTransitionChain(
-        // returns stateHashes array if valid
-        // else, reverts
+    function _bindSigsToVariableParts(
```

Comment:
I think we can avoid the need for this: the actual contract API should accept signed variable parts, and these can be constructed off chain (in the client code).

Reply:
Agree on this. Moreover, the actual contract API accepting signed variable parts would make it even more intuitive and convenient.